fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/OxidizableBlocksRegistry.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.registry;
+
+import java.util.Optional;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Oxidizable;
+import net.minecraft.item.HoneycombItem;
+
+import net.fabricmc.fabric.api.util.OxidizableFamily;
+
+public class OxidizableBlocksRegistry {
```

Comment:
I'd recommend letting people query the vanilla functions in `Oxidizable`. I'd only expose this API to inject into the relevant vanilla maps, which can be done by injecting into the synthetic lambda bodies: for example you can inject at the return of `method_34740` to replace the `ImmutableBiMap` by a regular `BiMap`, and later on append stuff to this `BiMap` directly.

Reply:
The `BiMap` returned by `method_34740` is immediately memoized, so adding stuff to it wouldn't do anything.

Reply:
> The `BiMap` returned by `method_34740` is immediately memoized

Haha, reminds me of how mojang creates a whole map every time you want to query the vanilla furnace fuel values

Reply:
Yes, the reference to the map is memorized. But you can replace the immutable map by a regular map and that mutable map will be memoized. You can then edit it later by just querying the supplier.

Reply:
Ah, yeah I see now. Alright, will do that then!